packages/firestore/test/unit/remote/serializer.helper.ts

```diff
@@ -0,0 +1,1487 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+
+import { Blob } from '../../../src/api/blob';
+import { PublicFieldValue as FieldValue } from '../../../src/api/field_value';
+import { GeoPoint } from '../../../src/api/geo_point';
+import { Timestamp } from '../../../src/api/timestamp';
+import { DocumentKeyReference } from '../../../src/api/user_data_reader';
+import { DocumentReference } from '../../../src/api/database';
+import { DatabaseId } from '../../../src/core/database_info';
+import {
+  ArrayContainsAnyFilter,
+  ArrayContainsFilter,
+  Direction,
+  FieldFilter,
+  InFilter,
+  KeyFieldFilter,
+  Operator,
+  OrderBy,
+  Query
+} from '../../../src/core/query';
+import { SnapshotVersion } from '../../../src/core/snapshot_version';
+import { Target } from '../../../src/core/target';
+import { TargetData, TargetPurpose } from '../../../src/local/target_data';
+import {
+  DeleteMutation,
+  FieldMask,
+  Mutation,
+  Precondition,
+  SetMutation,
+  VerifyMutation
+} from '../../../src/model/mutation';
+import { DOCUMENT_KEY_NAME, FieldPath } from '../../../src/model/path';
+import { refValue } from '../../../src/model/values';
+import * as api from '../../../src/protos/firestore_proto_api';
+import { JsonProtoSerializer } from '../../../src/remote/serializer';
+import {
+  DocumentWatchChange,
+  WatchTargetChange,
+  WatchTargetChangeState
+} from '../../../src/remote/watch_change';
+import { Code, FirestoreError } from '../../../src/util/error';
+import { forEach } from '../../../src/util/obj';
+import { addEqualityMatcher } from '../../util/equality_matcher';
+import {
+  bound,
+  byteStringFromString,
+  dbId,
+  deletedDoc,
+  deleteMutation,
+  doc,
+  field,
+  filter,
+  key,
+  orderBy,
+  patchMutation,
+  path,
+  ref,
+  setMutation,
+  testUserDataReader,
+  testUserDataWriter,
+  transformMutation,
+  version,
+  wrap,
+  wrapObject
+} from '../../util/helpers';
+
+import { ByteString } from '../../../src/util/byte_string';
+
+const userDataWriter = testUserDataWriter();
+const protobufJsonReader = testUserDataReader(/* useProto3Json= */ true);
+const protoJsReader = testUserDataReader(/* useProto3Json= */ false);
+
+/**
+ * Runs the serializer test with an optional ProtobufJS verification step
+ * (only provided in Node).
+ *
+ * These tests are initialized in 'serializer.browser.test.ts' and
+ * 'serializer.node.test.ts'.
+ */
+export function serializerTest(
```

Comment:
Note that this file is the same as `serializer.test.ts`, but wrapped in this function. I also pulled out `verifyProtobufJsRoundTrip`